### PR TITLE
Added prepack script

### PR DESCRIPTION
**Pull request by VictorLeach96**
This can be used to code sign binaries before they are compressed.

### CODE REVIEW

File: test/native-module-gyp/prepack.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+var fs = require('fs')
+var assert = require('assert')
+assert.equal(fs.existsSync('./build/Release/native.node'), true)
```
Comment (ralphtheninja) [inline test/native-module-gyp/prepack.js]: :+1:

**Comment by VictorLeach96** [comment]
Any more feedback on this? it's the last piece of the puzzle for our build pipeline

File: util.js [hunk 2]
```diff
@@ -23,13 +24,28 @@ function spawn (cmd, args, cb) {
...
+
 function exec (cmd, cb) {
   return execSpawn(cmd, {stdio: 'inherit'}).on('exit', function (code) {
     if (code === 0) return cb()
     cb(error.spawnFailed(cmd, [], code))
   })
 }
 
+function run (item, cb) {
+  if (item.substr(-3) === '.js' && fs.existsSync(item)) {
```
Comment (vweevers) [inline util.js]: Why check for file existence?

Comment (vweevers) [reply util.js]: Also, prefer `path.extname(item) === '.js'`

Comment (ralphtheninja) [reply util.js]: Why the `.fork()` in the first place? Doesn't `.exec()` work?

**Comment by VictorLeach96** [comment]
Great thanks :)

**Comment by vweevers** [comment]
Just a thought, we could consider supporting [subarg](https://github.com/substack/subarg) in the future. E.g.

```
prebuild --backend [ cmake-js --prefer-clang ] --prepack [ codesign -v -s MyCompany ]
```

**Comment by ralphtheninja** [comment]
7.6.0

**Comment by VictorLeach96** [comment]
Huge thanks to everyone for these! Massively helpful

**Pull request merged**